libr/core/graph.c

```diff
@@ -2218,6 +2218,28 @@ R_API RAGraph *r_agraph_new(RConsCanvas *can) {
 	return g;
 }
 
+static void visual_offset (RCore *core) {
+	char buf[256];
+	r_line_set_prompt ("[offset]> ");
+	strcpy (buf, "s ");
+	if (r_cons_fgets (buf+2, sizeof (buf)-3, 0, NULL) >0) {
+		if (buf[2]=='.')buf[1]='.';
+		r_core_cmd0 (core, buf);
+	}
+}
+
+static void print_agraph (RAGraph *g, RCore *core, struct agraph_refresh_data *grd){
```

Comment:
This function looks very similar to agraph_refresh, which is the function that is called on each iteration of the while() loop. So, probably we can merge this with agraph_refresh. In this way you just need to change seek with r_core_seek and the graph automatically updates itself